Mono.Addins/Mono.Addins.Database/AddinUpdateData.cs

```diff
@@ -185,6 +187,21 @@ public void RegisterExtension (AddinDescription description, ModuleDescription m
 					monitor.ReportWarning ("The add-in '" + description.AddinId + "' is trying to register the class '" + extension.Path.Substring (1) + "', but there isn't any add-in defining a suitable extension point");
 			}
 		}
+
+		static IEnumerable<ExtensionNodeDescription> GetNodesIgnoringConditions (Extension extension)
+		{
+			foreach (ExtensionNodeDescription node in extension.ExtensionNodes) {
+				if (node.IsCondition) {
+					//first node in a complex condition is the actual condition
+					bool skipFirst = node.NodeName == "ComplexCondition";
```

Comment:
Good catch. Yay for code reviews!

I suppose I should really have added a test but I wasn't sure of a good way and didn't want to stall merging while I did so.